### ISSUE TITLE
Maven plugin feature

### ISSUE THREAD

**Issue by xdevroey** [STAMP-project/botsing#4]
#### Characteristics
- **Issue Type**: feature

#### Description

Hi @luandrea @danzone !

Similar to dspot and descartes, I have created a maven sub-module dedicated to the botsing plugin. The main to call botsing is available [here](https://github.com/STAMP-project/botsing/blob/master/botsing-reproduction/src/main/java/eu/stamp/botsing/Main.java ) and the command line parameters are listed [here](https://github.com/STAMP-project/botsing/blob/master/botsing-reproduction/src/main/java/eu/stamp/botsing/CommandLineParameters.java). As you can see, we have drastically simplified the thing ;)

Please do a branch (or fork) and use a pull request for the Maven plugin. I will add a developer guide in the readme of the project. Please take a look at it. 

Best,

Xavier

**Comment by xdevroey** [comment]
The main change is: EvoSuite is now an external dependency we rely on in the botsing-reproduction module. 

I do not know how logging works with Maven, but we use Slf4J for logging so any compatible logging implementation should work. 

Here is an [example of usage}(https://github.com/STAMP-project/botsing/blob/master/botsing-reproduction/src/test/java/eu/stamp/botsing/SimpleCrashReproductionTest.java) (I asked Pouria to remove that from the repo, so here is a copy just in case):

```Java
public class SimpleCrashReproductionTest {

    public static int target_frame_level = 3;

    public static String user_dir = System.getProperty("user.dir");
    public static String test_dir = Paths.get(user_dir,"src","test","java","eu","stamp","botsing").toString();
    public static String log_dir = Paths.get(test_dir, "sample.log").toString();
    public static String  bin_path = Paths.get(test_dir, "sample_dep").toString();
    private static String separator = System.getProperty("path.separator");

    @Test
    public void runtest(){
        setLoggingLevel(Level.INFO);

        BotSing botsing = new BotSing();
        String[] prop = {
                "-Dcrash_log="+log_dir,
                "-Dtarget_frame="+target_frame_level,
                "-projectCP",
                getListOfDeps(),
        };
        botsing.parseCommandLine(prop);
    }

    public static void setLoggingLevel(ch.qos.logback.classic.Level level) {
        ch.qos.logback.classic.Logger root = (ch.qos.logback.classic.Logger) org.slf4j.LoggerFactory.getLogger(ch.qos.logback.classic.Logger.ROOT_LOGGER_NAME);
        root.setLevel(level);
    }

    public static String getListOfDeps(){
        String dependencies = "";
        File depFolder = new File(bin_path);
        File[] listOfFilesInSourceFolder = depFolder.listFiles();
        for(int i = 0; i < listOfFilesInSourceFolder.length; i++) {
            if (listOfFilesInSourceFolder[i].getName().charAt(0) != '.') {
                Path depPath = Paths.get(depFolder.getAbsolutePath(), listOfFilesInSourceFolder[i].getName());
                dependencies += (depPath.toString() + separator);
            }
        }
        return dependencies;
    }
}
```

**Comment by pderakhshanfar** [comment]
In addition to the information that Xavier provided, there is a config file in Botsing that you can use it to set the optional parameters of the search process. This config file is [here](https://github.com/STAMP-project/botsing/blob/master/botsing-reproduction/src/main/java/eu/stamp/botsing/config.properties).

**Comment by danzone** [comment]
Hi @xdevroey @pderakhshanfar , excellent!!! Today we'll start to inspect the code in order to schedule our development activity!

**Comment by luandrea** [comment]
Hi @xdevroey @pderakhshanfar,
I took a look at the code and tried to configure Botsing through the Maven properties, but I saw that Botsing reads only crash_log and target_frame properties from command line, plus some other properties from ${"user.dir"}/src/main/java/eu/stamp/botsing/config.properties .
I need to configure all properties used by Botsing from my class, What do you suggest to do?

**Comment by luandrea** [comment]
Hi @xdevroey @pderakhshanfar,
I pushed the first version of the maven plugin in my forked repository, but I cannot go on without resolving the problems that I pointed out in the previous comment. Could you take a look at them?

**Comment by luandrea** [comment]
I will add them as separate feature to be implemented to better track their status.

**Comment by xdevroey** [comment]
Merged pull request #17 in commit c123f111bc60379670647e9ecb3e592198ba7c2e
